C/primitive/elements/jets.h

```diff
@@ -85,19 +85,23 @@ bool output_scripts_hash(frameItem* dst, frameItem src, const txEnv* env);
 bool output_range_proofs_hash(frameItem* dst, frameItem src, const txEnv* env);
 bool output_surjection_proofs_hash(frameItem* dst, frameItem src, const txEnv* env);
```

Comment:
Let's add `output_surjection_proof_hash`. The `~proofs_hash` is part of `tx_hash` so it seems we need the surjection proof hash to fully commit to an output.

Reply:
The TL;DR is that surjection proofs depend on both the inputs and outputs and thus are not appropriate for output only sighash operations.

The longer story is that both the output range proofs and surjection proofs are not covered by the signature at all in legacy sighashes, since they are basically of fixed size and, as proofs, are not really supposed to have information in any information theoretic sense.  In element's taproot deployment the output range proofs and surjection proofs were added to the sighash in order to to cover all bases.  However we didn't think too much about this at the time and it has resulted in complaints about certain sighash modes.

As I noted above the surjection proofs depend on the inputs, and sometimes when users want to sign in some output only modes, the inputs needed in order to generate the surjection proofs are not yet available, making some tasks impossible to do with taproot based signatures due to the inclusion of surjection proofs in those sighashes.

Reply:
This probably needs to be documented somewhere, but I'm not really sure where.  It is kinda of a quirky Element's thing.

Reply:
The Elements' sighashes differ from Bitcoin's in interesting ways. I will write some documentation in my https://github.com/uncomputable/bitcoin-notes repo. We should add a comment to the output-surjection-proofs jet explaining why it doesn't make sense for single outputs.

Fair point that a ZKP is random noise from an information theoretic sense. The ZKP is malleable if it is not covered by the signature. However, it is exponentially unlikely that someone will fake the proof.